frontend/app/controllers/TierController.scala

```diff
@@ -43,19 +42,30 @@ trait UpgradeTier {
 
   def upgrade(tier: Tier) = MemberAction.async { implicit request =>
 
-    def futureCustomerOpt = request.member match {
+    def futurePaidPreviewOpt = request.member match {
       case paidMember: PaidMember =>
-        request.touchpointBackend.stripeService.Customer.read(paidMember.stripeCustomerId).map(Some(_))
+        for {
+          preview <- MemberService.previewUpgradeSubscription(paidMember, request.user, tier)
+          customer <- request.touchpointBackend.stripeService.Customer.read(paidMember.stripeCustomerId)
+        } yield Some(PaidPreview(customer.card, preview))
       case _: FreeMember => Future.successful(None)
     }
 
+    val subscriptionService = request.touchpointBackend.subscriptionService
+
     if (request.member.tier < tier) {
       for {
-        customerOpt <- futureCustomerOpt
+        paidPreviewOpt <- futurePaidPreviewOpt
+        subscriptionStatus <- subscriptionService.getSubscriptionStatus(request.member)
+        currentSubscription <- subscriptionService.getSubscriptionDetails(subscriptionStatus.current)
```

Comment:
@jennysivapalan same problem with Scala future for-comprehensions is going on here - they're executing in series